src/openms/source/KERNEL/BaseFeature.cpp

```diff
@@ -191,18 +188,84 @@ namespace OpenMS
           seqs.insert(id_tmp.getHits()[0].getSequence().toString());
         }
       }
-      if (seqs.size() == 1)
-      {
-        return FEATURE_ID_MULTIPLE_SAME; // hits have identical seqs
-      }
-      else if (seqs.size() > 1)
-      {
-        return FEATURE_ID_MULTIPLE_DIVERGENT;                        // multiple different annotations ... probably bad mapping
-      }
-      else /*if (seqs.size()==0)*/
+      if (seqs.size() == 1) return FEATURE_ID_MULTIPLE_SAME; // hits have identical seqs
+      if (seqs.size() > 1)
+        return FEATURE_ID_MULTIPLE_DIVERGENT; // multiple different annotations ... probably bad mapping
+      else /*if (seqs.size()==0)*/ return FEATURE_ID_NONE;   // very rare case of empty hits
+    }
+    else // consider IDs in new format
+    {
+      if (id_matches_.size() == 1) return FEATURE_ID_SINGLE;
+      // if there are multiple IDs, check if all are equal (to the first):
+      auto it = id_matches_.begin();
+      IdentificationData::IdentifiedMolecule molecule = (*it)->identified_molecule_var;
+      for (++it; it != id_matches_.end(); ++it)
       {
-        return FEATURE_ID_NONE;   // very rare case of empty hits
+        if ((*it)->identified_molecule_var != molecule) return FEATURE_ID_MULTIPLE_DIVERGENT;
       }
+      return FEATURE_ID_MULTIPLE_SAME;
+    }
+  }
+
+
+  bool BaseFeature::hasPrimaryID() const
+  {
+    return bool(primary_id_);
+  }
+
+
+  const IdentificationData::IdentifiedMolecule& BaseFeature::getPrimaryID() const
+  {
+    if (!primary_id_)
+    {
+      throw Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION,
+                                          "no primary ID assigned");
+    }
+
+    return *primary_id_; // unpack the option
+  }
+
+
+  void BaseFeature::clearPrimaryID()
+  {
+    primary_id_ = boost::none;
+  }
+
+
+  void BaseFeature::setPrimaryID(const IdentificationData::IdentifiedMolecule& id)
+  {
+    primary_id_ = id;
+  }
+
+
+  const std::set<IdentificationData::ObservationMatchRef>& BaseFeature::getIDMatches() const
+  {
+    return id_matches_;
+  }
+
+
+  std::set<IdentificationData::ObservationMatchRef>& BaseFeature::getIDMatches()
+  {
+    return id_matches_;
+  }
+
+
+  void BaseFeature::addIDMatch(IdentificationData::ObservationMatchRef ref)
+  {
+    id_matches_.insert(ref);
+  }
+
+  void BaseFeature::updateIDReferences(const IdentificationData::RefTranslator& trans)
+  {
+    if (primary_id_ != boost::none)
```

Comment:
Just based on this context I don't have a good intuition what it means of heaving this optional parameter
and what the translation does. 
can you add a comment here to guide the reader in the right direction?